added fault-failure codes from excel sheet

ipa-hsd opened the pull request:
Auto-generated from `gdocs_reader.py`  

Note: Not sure if the excel sheet is finalized, can keep this open till then

ChrisTimperley commented on mavros/ff581a0/ff581a0.bug:
```diff
@@ -60,3 +60,9 @@ bugzoo:
+fault-codes:
+  - MODELS:ROBOT
+  - MODELS:TRANSFORMATIONS
+failure-codes:
```
From a quick glance, it like the `fault-codes` and `failure-codes` are identical to one another for all of the bugs?

ipa-hsd replied:
you are right, var mix-up in the script.

ipa-hsd commented:
removed the fault failure codes, since there some issues still. Changes done in (copy of) excel sheet;

- Changed all labels to uppper case
- fixed spaces in labels

Missing bugs in excel sheet
`geometry2/3ec0485/3ec0485.bug` 

`PROGRAMMING` (-only) fault label
`geometry2/1b44c4d/1b44c4d.bug` --> `BDO:BUILD`
`geometry2/1b5fa94/1b5fa94.bug` --> `PROGRAMMING:RESOURCES`
`geometry2/2153716/2153716.bug` --> `BDO:BUILD`

`geometry2/164cfa3/164cfa3.bug` (`SYSTEM?` --> removed)
`geometry2/439e235/439e235.bug` (`SOFTWARE:SILENT(?)` --> removed)

Empty fields 
`geometry2/c81fa72/c81fa72.bug` (failure-code --> `SYSTEM:UNINTENDED-BEHAVIOUR`)
`turtlebot/928306b/928306b.bug` (failure-code --> `SOFTWARE:UX`, `SYSTEM:NONE`)

ipa-hsd commented:
Errors detected by validator:
```
failure-codes.0: 'SOFWARE:BUILD' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/3a48064/3a48064.bug

failure-codes.1: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/e122ccc/e122ccc.bug

ERROR: Empty fields encountered for kobuki/496aac8/496aac8.bug in 'ROBUST coding'
failure-codes: 'None' is not a list.
Error validating bug description kobuki/496aac8/496aac8.bug

failure-codes.0: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description motoman/eadbcb8/eadbcb8.bug

fault-codes.1: 'MODEL:ROBOT' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description motoman/2d42582/2d42582.bug

failure-codes.0: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/cd273dd/cd273dd.bug

fault-codes.0: 'MODEL:ROBOT' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description universal_robot/89145c4/89145c4.bug

failure-codes.0: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/18ffa89/18ffa89.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/263650d/263650d.bug

fault-codes.0: 'EVOLUTION:FILES' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description kobuki/5abe7d4/5abe7d4.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description care-o-bot/43705f7/43705f7.bug

failure-codes.1: 'SYSTEM:PHYSICAL?' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description kobuki/3c4c399/3c4c399.bug

fault-codes.0: 'EVOLUTION:INTERNAL' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description geometry2/0481047/0481047.bug

failure-codes.1: 'SOFWARE:RUNTIME' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/164cfa3/164cfa3.bug

failure-codes.0: 'SOFWARE:RUNTIME' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/561d66b/561d66b.bug

fault-codes.0: 'PROGRAMMING:NAMESPACE' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description geometry2/b206807/b206807.bug

failure-codes.1: 'SYSTEM:UNINTENDED_BEHAVIOUR (EXTRAPOLATING)' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description confidential/86cb680/86cb680.bug

fault-codes.0: 'EVOLUTION:FILES' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description mavros/27e7db9/27e7db9.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/84264f0/84264f0.bug

failure-codes.2: 'SYSTEM:MOTION?' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/de2cc36/de2cc36.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/e1a8005/e1a8005.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/ff581a0/ff581a0.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description other/473ab5e/473ab5e.bug

fault-codes.1: 'CONFIG:' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description turtlebot/928306b/928306b.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description turtlebot/9ffffca/9ffffca.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/1e0904c/1e0904c.bug

fault-codes.1: 'PROGRAMMING:UNITIALISED' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description geometry2/d50d51b/d50d51b.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/da141a5/da141a5.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/ede2646/ede2646.bug

fault-codes.0: 'PROGRAMMING:SYNTAX-ERROR' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description geometry2/f19569c/f19569c.bug

fault-codes.0: 'EVOLUTION:ROBOT?' not in ('BAD-SMELL', 'ROS-SPECIFIC', 'NAMING-TYPOS', 'COPYPASTA', 'CONFIG:PARAM', 'CONFIG:REMAPPING', 'CONFIG:ARGS', 'CONFIG:CONSTANTS', 'CONFIG:ENCODING', 'CONFIG:NAMESPACE', 'CONFIG:TOPIC', 'BDO:BUILD', 'BDO:DEPENDENCY', 'BDO:ORCHESTRATION', 'CONCURRENCY:NO-SYNC', 'CONCURRENCY:BAD-SYNC', 'CONCURRENCY:TIMING', 'CONCURRENCY:SIGNALS', 'EVOLUTION:LANGUAGE', 'EVOLUTION:LIBRARY', 'EVOLUTION:PACKAGE', 'EVOLUTION:DEPRECATION', 'EVOLUTION:ROBOT', 'EVOLUTION:FIRMWARE', 'EVOLUTION:DOCS', 'EVOLUTION:PATHS', 'PROGRAMMING:LOGIC', 'PROGRAMMING:CALCULATIONS', 'PROGRAMMING:CONTROL-FLOW', 'PROGRAMMING:MISSING-FEATURE', 'PROGRAMMING:VALIDATION', 'PROGRAMMING:UNINITIALISED', 'PROGRAMMING:BROKEN-CONTRACT', 'PROGRAMMING:RESOURCES', 'PROGRAMMING:DATATYPE', 'PROGRAMMING:UNUSED', 'PROGRAMMING:STRING-FORMATTING', 'MODELS:ROBOT', 'MODELS:WORLD', 'MODELS:TRANSFORMATIONS', 'SYSTEMS:OS', 'SYSTEMS:HARDWARE', 'SYSTEM:FIRMWARE', 'SYSTEMS:CONFIG')
Error validating bug description kobuki/b18f559/b18f559.bug

failure-codes.1: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description kobuki/493e3f9/493e3f9.bug

failure-codes.2: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description kobuki/4115400/4115400.bug

failure-codes.0: 'SYSTEM:SIMULATION' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/cda133d/cda133d.bug

```

ipa-hsd commented:
@ChrisTimperley please review the python script `bug_validator.py`. It might be useful later to validate `commits` and other subsections individually.

ipa-hsd commented on universal_robot/56cf07f/56cf07f.bug:
```diff
@@ -73,3 +73,8 @@ fix:
+  - BDO:DEPENDENCY
+failure-codes:
+  - SOFTWARE:BUILD
+  - SYSTEM:NONE
```
How did this pass the `yamllint` test before?

ipa-hsd replied:
From [here](https://travis-ci.com/github/robust-rosin/robust/builds/216629561)
```bash
./universal_robot/56cf07f/56cf07f.bug:31:81: [warning] line too long (88 > 80 characters) (line-length)
./universal_robot/56cf07f/56cf07f.bug:34:81: [warning] line too long (83 > 80 characters) (line-length)
./universal_robot/56cf07f/56cf07f.bug:36:81: [warning] line too long (85 > 80 characters) (line-length)
./universal_robot/56cf07f/56cf07f.bug:37:81: [warning] line too long (86 > 80 characters) (line-length)
./universal_robot/56cf07f/56cf07f.bug:38:81: [warning] line too long (116 > 80 characters) (line-length)
```
No changes were done to these lines here

ipa-hsd commented:
Pending issues related to `SOFTWARE:SILENT`
```
failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/263650d/263650d.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description care-o-bot/43705f7/43705f7.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/84264f0/84264f0.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/e1a8005/e1a8005.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description mavros/ff581a0/ff581a0.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description other/473ab5e/473ab5e.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description turtlebot/9ffffca/9ffffca.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description universal_robot/1e0904c/1e0904c.bug

failure-codes.1: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/da141a5/da141a5.bug

failure-codes.0: 'SOFTWARE:SILENT' not in ('WARNING', 'UNKNOWN', 'HARMLESS', 'MISLEADING', 'NONE', 'SYSTEM:LIVENESS', 'SYSTEM:UNINTENDED-BEHAVIOUR', 'SYSTEM:PERFORMANCE', 'SYSTEM:NONE', 'SYSTEM:PHYSICAL', 'SYSTEM:MOTION', 'SYSTEM:SIMULATION', 'SOFTWARE:BUILD', 'SOFTWARE:RUNTIME', 'SOFTWARE:NETWORK', 'SOFTWARE:UX', 'SOFTWARE:CONCURRENCY', 'SOFTWARE:PERFORMANCE')
Error validating bug description geometry2/ede2646/ede2646.bug

```

ipa-hsd commented:
The frequency of fault-failure codes calculated in the Juypter notebook: https://github.com/robust-rosin/robust/issues/387

ChrisTimperley commented on lib/robust/robust.py:
```diff
@@ -235,6 +235,30 @@ def commits(self) -> t.AbstractSet[FixCommit]:
+    def faults(self) -> t.Optional[t.List[str]]:
+        return self._get_value('fault-codes')
+
+    def failures(self) -> t.Optional[t.List[str]]:
```
missing `@property`?

ChrisTimperley commented on kobuki/2f647af/2f647af.bug:
```diff
@@ -71,3 +71,8 @@ fix:
+fault-codes:
+  - PROGRAMMING:CALCULATIONS
+failure-codes:
+  - SYSTEM:UNINTENDED-BEHAVIOUR
```
There doesn't seem to be a SOFTWARE failure in here

ChrisTimperley commented on kobuki/1c141a5/1c141a5.bug:
```diff
@@ -41,3 +41,9 @@ fix:
+  - PROGRAMMING:DATATYPE
+  - PROGRAMMING:CALCULATIONS
+failure-codes:
+  - SYSTEM:UNINTENDED-BEHAVIOUR
```
No software-level failure

ipa-hsd commented:
@ChrisTimperley ping. Would like this one to be merged before https://github.com/robust-rosin/robust/pull/399 as I want to update the schema from here.

ChrisTimperley reviewed (approved):
LGTM 👍🏻
